controllers/routemonitor/routemonitor_controller_supplement.go

```diff
@@ -0,0 +1,302 @@
+	labels := commonTemplateLables()
+	// hardcode the replicasize for no
+	//replicas := m.Spec.Size
+	var replicas int32 = 1
+
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackBoxName,
+			Namespace: blackBoxNamespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image: "prom/blackbox-exporter:master",
+						Name:  "blackbox-exporter",
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: blackBoxPortNumber,
+							Name:          blackBoxPortName,
+						}},
+					}},
+				},
+			},
+		},
+	}
+	// Set BlackboxExporter instance as the owner and controller
+	// ctrl.SetControllerReference(m, dep, r.Scheme)
+	return dep
+}
+
+// templateForBlackBoxExporterService returns a blackbox service
+func (r *RouteMonitorReconciler) templateForBlackBoxExporterService() *corev1.Service {
+	labels := commonTemplateLables()
+
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackBoxName,
+			Namespace: blackBoxNamespace,
+			Labels:    labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: labels,
+			Ports: []corev1.ServicePort{{
+				TargetPort: intstr.FromString(blackBoxPortName),
+				Port:       blackBoxPortNumber,
+				Name:       blackBoxPortName,
+			}},
+		},
+	}
+	// Set BlackboxExporter instance as the owner and controller
+	// ctrl.SetControllerReference(m, dep, r.Scheme)
+	return svc
+}
+
+// templateForServiceMonitorResource returns a ServiceMonitor
+func (r *RouteMonitorReconciler) templateForServiceMonitorResource(routeMonitor *v1alpha1.RouteMonitor, serviceMonitorName string) *monitoringv1.ServiceMonitor {
+
+	routeURL := routeMonitor.Status.RouteURL
+
+	routeMonitorLabels := commonTemplateLables()
+	var labelSelector = metav1.LabelSelector{}
+	err := metav1.Convert_Map_string_To_string_To_v1_LabelSelector(&routeMonitorLabels, &labelSelector, nil)
+	if err != nil {
+		r.Log.Error(err, "Failed to convert LabelSelector to it's components")
+	}
+	// Currently we only support `http_2xx` as module
+	// Still make it a variable so we can easily add functionality later
+	modules := []string{"http_2xx"}
+
+	params := map[string][]string{
+		"module": modules,
+		"target": {routeURL},
+	}
+
+	serviceMonitor := &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: serviceMonitorName,
+			// ServiceMonitors need to be in `openshift-monitoring` to be picked up by cluster-monitoring-operator
+			Namespace: blackBoxNamespace,
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			JobLabel: serviceMonitorName,
+			Endpoints: []monitoringv1.Endpoint{
+				{
+					// TODO use variable from blackbox exporter deployment
```

Comment:
I don't understand this TODO, this port is set at the service as well, isn't it?

Reply:
I think that's an artifact from the code i initially hacked together